python_modules/libraries/dagster-dbt/dagster_dbt/dbt_project.py

```diff
@@ -48,16 +45,17 @@ def __init__(
         if not self.project_dir.exists():
             raise DagsterDbtProjectNotFoundError(f"project_dir {project_dir} does not exist.")
         self._target_folder = Path(target_folder)
-        self._prepare_command = prepare_command
-        # this is ok if it doesn't exist, will get created by `prepare`
+        # this is ok if it doesn't exist, will get created during sync
         self._packaged_project_dir = Path(packaged_project_dir) if packaged_project_dir else None
 
         if self._should_use_built_artifacts():
             self._manifest_path = self._current_project_dir.joinpath(
                 self._target_folder, "manifest.json"
             )
         else:
-            self._manifest_path = self._prepare_manifest()
+            self._manifest_path = self.generate_artifacts(
+                log_level=logging.DEBUG,  # quiet logging for init time generation
+            )
```

Comment:
We should do some assertion here that `self._manifest_path == expected_manifest_path` in either case. That way, we can ensure that `self.target_folder` was used properly if the user decides to override this function.

```
            expected_manifest_path = self._current_project_dir.joinpath(
                self._target_folder, "manifest.json"
            )
            self._manifest_path = self.generate_artifacts(
                log_level=logging.DEBUG,  # quiet logging for init time generation
            )
            
            check.invariant(self._manifest_path == expected_manifest_path)
            
```